### PR TITLE
Report Strict Accuracy Metric

### PR DESCRIPTION
**Pull request by ashors1**
Adds option to report “strict accuracy” metric, which requires full sequence matching. This is needed for evaluation using the Lambada dataset.

### CODE REVIEW

**Comment by zhangqiaorjc** [comment]
@ashors1 could you run linter to remove extra whitespace?

**Comment by zhangqiaorjc** [comment]
@laurentes asked is it sufficient to just set `apply_eval_sample_weights = True` instead of having this PR? @ashors1 i think you might need to motivate this PR a little bit more?

**Comment by ashors1** [comment]
When performing evaluation on the Lambada dataset, the metric we are typically interested in is "strict accuracy". In contrast to standard evaluation accuracy, which gives credit for partial word matching, strict accuracy requires that the full word be predicted correctly. Standard evaluation accuracy will be an overestimate of the true proportion of words predicted correctly.

File: praxis/layers/models.py [hunk 1]
```diff
@@ -98,6 +99,8 @@ def compute_xent_loss_helper(
       example weights from the input `eval_sample_weights` or not. When enabled,
       these per-example weights will be merged with the per token
       `input_batch.weights`.
+    report_strict_acc: Whether to report strict accuracy. Used for eval on
```
Comment (laurentes) [inline praxis/layers/models.py:102]: Could you extend this comment? In the case of Lambada, I guess you're only expecting a single word. But what this metrics seems to do is compute a per sequence exact match at least for the non-zero weighted tokens.

File: praxis/layers/models.py [hunk 2]
```diff
@@ -136,6 +151,9 @@ def compute_xent_loss_helper(
       fraction_of_correct_next_step_preds=(mean_acc, metric_weight),
       num_predictions=(num_preds, jnp.array(1.0, num_preds.dtype)),
   )
+  if report_strict_acc:
```
Comment (laurentes) [inline praxis/layers/models.py:154]: Could you move the new computation under this conditional too? It feels awkward to do the computation no matter what, but only add it to the metrics if desired.

**Review by laurentes — COMMENTED** [review]
Thanks Anna! Just a couple of suggestions before I approve.
